IOPool/Input/src/RootTree.cc

```diff
@@ -549,18 +605,13 @@ namespace edm {
                                            unsigned int cacheSize,
                                            char const* branchNames) {
       tree->LoadTree(0);
-      tree->SetCacheSize(cacheSize);
-      std::unique_ptr<TTreeCache> treeCache(dynamic_cast<TTreeCache*>(file.GetCacheRead()));
+      std::unique_ptr<TTreeCache> treeCache = file.createCacheWithSize(*tree, cacheSize);
       if (nullptr != treeCache.get()) {
         treeCache->StartLearningPhase();
         treeCache->SetEntryRange(0, tree->GetEntries());
         treeCache->AddBranch(branchNames, kTRUE);
         treeCache->StopLearningPhase();
       }
```

Comment:
I'm wondering the the `treeCache` should be active in the `TFile` during these `treeCache` calls (also elsewhere).

Reply:
According to @pcanal `treeCache` does not need to be active in the `TFile` during these calls.